src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala

```diff
@@ -100,11 +102,34 @@ object TezosTypes {
     signature: Option[String]
   )
 
+  sealed trait BlockMetadata extends Product with Serializable
+
+  final case object GenesisMetadata extends BlockMetadata
 
   final case class BlockHeaderMetadata(
-    balance_updates: Option[List[OperationMetadata.BalanceUpdate]]
+    balance_updates: List[OperationMetadata.BalanceUpdate],
+    nonce_hash: Option[NonceHash],
```

Comment:
just wondering possibly for future improvement, how much work would be to make circe decode/encode if we used camel case in the case classes instead of snake case. Would be it worth it? what do you think?

Reply:
I think it shouldn't be that hard. I didn't check, but I can imagine that the implicit decoding configuration can be simply tailored to adapt the casing (if it doesn't already by itself).

I'm not sure we might want to do that, tho, as I find it very convenient to have the same exact field naming between json and scala.
I've often found that mismatches in casing rules actually make things uselessly harder to work on, only for the sake of conventions that are not necessarily useful.